standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/PropertyServlet.java

```diff
@@ -331,41 +313,49 @@ private void runGet(HttpServletRequest request,
     }
   }
 
+  /**
+   * Single servlet creation helper.
+   */
+  private static class ServerBuilder extends ServletServerBuilder {
+    final int port;
+    final String path;
+    ServerBuilder(Configuration conf) {
+      super(conf);
+      port = MetastoreConf.getIntVar(conf, MetastoreConf.ConfVars.PROPERTIES_SERVLET_PORT);
+      path = MetastoreConf.getVar(conf, MetastoreConf.ConfVars.PROPERTIES_SERVLET_PATH);
+    }
+
+    @Override
+    protected String getServletPath() {
+      return path;
+    }
+
+    @Override
+    protected int getServerPort() {
+      return port;
+    }
+
+    @Override
+    protected HttpServlet createServlet() throws IOException {
```

Comment:
`IOException` is not thrown here.
IDE shows that it is not needed here and I verified that it compiles successfully without it.